packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/MapperUtils.java

```diff
@@ -128,19 +128,21 @@ public static IType createEnumType(ChoiceSchema enumType, boolean expandable, bo
     public static IType getExpectedResponseBodyType(Operation operation, JavaSettings settings) {
         final Schema responseSchema = SchemaUtil.getLowestCommonParent(operation.getResponseSchemas().iterator());
         if (responseSchema != null && responseSchema.isXmlWrapped()) {
-            // Create and return type for the XML wrapped schema.
-            //
-            // Note: XML wrapped response schemas are defined as ArraySchema but in reality it's a specialized
-            // ObjectSchema.
-            final ArraySchema arraySchema = (ArraySchema) responseSchema;
-            final String className = arraySchema.getElementType().getLanguage().getJava().getName() + "Wrapper";
-            final String classPackage = settings.isCustomType(className)
-                ? settings.getPackage(className)
-                : settings.getPackage(settings.getImplementationSubpackage() + ".models");
-            return new ClassType.Builder().packageName(classPackage)
-                .name(className)
-                .extensions(responseSchema.getExtensions())
-                .build();
+            if (responseSchema instanceof ArraySchema) {
+                // Create and return type for the XML wrapped schema.
+                //
+                // Note: XML wrapped response schemas are defined as ArraySchema but in reality it's a specialized
+                // ObjectSchema.
+                final ArraySchema arraySchema = (ArraySchema) responseSchema;
+                final String className = arraySchema.getElementType().getLanguage().getJava().getName() + "Wrapper";
+                final String classPackage = settings.isCustomType(className)
+                    ? settings.getPackage(className)
+                    : settings.getPackage(settings.getImplementationSubpackage() + ".models");
+                return new ClassType.Builder().packageName(classPackage)
+                    .name(className)
+                    .extensions(responseSchema.getExtensions())
+                    .build();
+            }
```

Comment:
TODO, this fails on xml test, as m4 process it differently.

As currently we are not able to test GET due to annotation processor bug, I am not sure whether this would be the correct code https://github.com/Azure/azure-sdk-for-java/issues/45885